lib/shim/message-shim.js

```diff
@@ -356,108 +354,37 @@ function recordConsume(nodule, properties, spec) {
-          callback: spec.callback,
-          promise: spec.promise,
-          messageHandler: spec.messageHandler
-        }
-
-        const destIdx = shim.normalizeIndex(args.length, spec.destinationName)
-        if (destIdx !== null) {
-          msgDesc.destinationName = args[destIdx]
-        }
-      }
-
-      // Make the segment if we can.
-      if (!msgDesc) {
-        shim.logger.trace('Not recording consume, no message descriptor.')
-        return fn.apply(this, args)
-      }
-
-      const name = _nameMessageSegment(shim, msgDesc, shim._metrics.CONSUME)
-
-      // Adds details needed by createSegment when used with a spec
-      msgDesc.name = name
-      msgDesc.recorder = genericRecorder
-      msgDesc.parent = parent
-
-      const segment = shim.createSegment(msgDesc)
-      const getParams = shim.agent.config.message_tracer.segment_parameters.enabled
-      const resHandler = shim.isFunction(msgDesc.messageHandler) ? msgDesc.messageHandler : null
-
-      const cbIdx = shim.normalizeIndex(args.length, msgDesc.callback)
-      if (cbIdx !== null) {
-        shim.bindCallbackSegment(args, cbIdx, segment)
-
-        // If we have a callback and a results handler, then wrap the callback so
-        // we can call the results handler and get the message properties.
-        if (resHandler) {
-          shim.wrap(args, cbIdx, function wrapCb(shim, cb, cbName) {
-            if (shim.isFunction(cb)) {
-              return function cbWrapper() {
-                const cbArgs = shim.argsToArray.apply(shim, arguments)
-                const msgProps = resHandler.call(this, shim, cb, cbName, cbArgs)
-                if (getParams && msgProps && msgProps.parameters) {
-                  shim.copySegmentParameters(segment, msgProps.parameters)
-                }
-
-                return cb.apply(this, arguments)
-              }
-            }
-          })
-        }
+  return this.record(nodule, properties, function consumeRecorder(shim, fn, fnName, args) {
+    if (shim.isFunction(spec)) {
+      spec = spec.apply(this, arguments)
+    } else {
+      spec = { ...DEFAULT_SPEC, ...spec }
+      const destIdx = shim.normalizeIndex(args.length, spec.destinationName)
+      if (destIdx !== null) {
+        spec.destinationName = args[destIdx]
       }
-
-      // Call the method in the context of our segment.
-      let ret = shim.applySegment(fn, segment, true, this, args)
-
-      if (ret && msgDesc.promise && shim.isPromise(ret)) {
-        ret = shim.bindPromise(ret, segment)
-
-        // Intercept the promise to handle the result.
-        if (resHandler) {
-          ret = ret.then(function interceptValue(res) {
-            const msgProps = resHandler.call(this, shim, fn, fnName, res)
-            if (getParams && msgProps && msgProps.parameters) {
-              shim.copySegmentParameters(segment, msgProps.parameters)
-            }
-            return res
-          })
+    }
+    const messageHandler = spec.messageHandler
```

Comment:
this is so weird, I don't see any of our instrumentation using this: sns, sqs, nor amqp.  We have tests for it though and technically this is a pubilc API but wow

Reply:
I can't find anyone on Github using it either, but maybe we have some secret fans out there.

Reply:
If we're making this a semver major should we just remove?

Reply:
I don't see a reason to break more than we have to. It's kind of convenient to to keep it around. Would be more work to remove it: would have to update our docs and examples.

Reply:
Wait, we do use it ourselves: we use it for amqplib.